docs/manual-testing-playbook.md

```diff
@@ -0,0 +1,997 @@
+You may see a note about "reading the system DNS configuration is not supported on this platform";
+this does not appear to be related to snap confinement, merely a system limitation.
+
+Now you can test resolving the Tailscale MagicDNS FQDN through the `tailscale dns query` subcommand,
+as well as the system resolver.
+
+```bash
+TAILNET_DOMAIN=$(lxc exec tailscale-1 -- sh -c "tailscale dns status | awk 'match(\$0, /\w+\.ts\.net/) {print substr(\$0, RSTART, RLENGTH); exit}'")
+lxc exec tailscale-1 -- tailscale dns query tailscale-2.$TAILNET_DOMAIN a
+lxc exec tailscale-1 -- dig tailscale-2.$TAILNET_DOMAIN
+```
+
+Both of the above queries should return the same ipv4 address: the tailnet address of `TAILSCALE_VM_2`.
+Resolvectl should confirm the system DNS settings applied by tailscale:
+
+```bash
+lxc exec tailscale-1 -- resolvectl
+```
+
+Note that DNS queries for the bare machine name don't work (no result returned):
+
+```bash
+lxc exec tailscale-1 -- tailscale dns query tailscale-2 a
+lxc exec tailscale-1 -- dig tailscale-2
+```
+
+This is unexpected, but does not appear to be related to snap confinement.
+
+### Tailscale SSH
+
+- Tailscaled in ssh mode (ie. run with `--ssh`) does not work (incoming ssh connections on the tailnet will not work, failing with permission denied).
+- `tailscale ssh` will work if sshing to a host running tailscaled, if that tailscaled is not installed via the strictly confined snap.
+
+### Tailscale update
+
+Tailscale includes a built in updater, which won't work because the snap files are immutable.
+The behaviour will look like this (if a newer version of Tailscale is available):
+
+```bash
+lxc exec tailscale-1 -- tailscale update --yes
+```
+
+Output:
+
+```text
+open /etc/apt/sources.list.d/tailscale.list: no such file or directory
+```
+
+If Tailscale is already at the latest version, you will simply see a message about no updates needed.
+
+## Cleanup
+
+To cleanup, you can remove the machines from the Tailscale web admin console, on the "Machines" tab.
+
+Then delete the LXD VMs:
+
+```bash
+lxc delete --force tailscale-1 tailscale-2
+```
+
+## Unknowns
+
+Untested cases that are not covered in this playbook:
+
+- `tailscale switch`: This requires multiple accounts. This is expected to work, since the login/logout sub commands work.
+- `tailscale configure kubeconfig`: This requires a Kubernetes cluster. This may not work if requires certain file paths, but otherwise may work.
+- `tailscale lock disablement-kdf`, `tailscale lock local-disable`, `tailscale lock revoke-keys`: These were not in scope of testing in this document, but they are expected to work.
+- Custom configuration options (arguments to `tailscale up`, `tailscale login`, or `tailscale set`): As noted in an earlier section, there are many options. `--ssh` is known to not work, but the others are expected to work.
```

Comment:
Also note that since I compiled the list of available tailscale subcommands, there are now more:

- `tailscale metrics` (prints prometheus metrics) - seems to work for me
- `tailscale syspolicy` ("tools for diagnosing the MDM and system policy configuration") - I'm not sure what this is

I also just discovered that there's an undocumented `debug` subcommand, with many subcommands of its own.

Reply:
Tracking in https://github.com/canonical/tailscale-snap/issues/37